src/data_source/storage/sql.rs

```diff
@@ -931,21 +1135,21 @@ mod test {
         setup_test();
 
         let db = TmpDb::init().await;
-        let port = db.port();
-        let host = &db.host();
+        let db_opt = db.config().db_opt;
 
-        let connect = |migrations: bool, custom_migrations| async move {
-            let mut cfg = Config::default()
-                .user("postgres")
-                .password("password")
-                .host(host)
-                .port(port)
-                .migrations(custom_migrations);
-            if !migrations {
-                cfg = cfg.no_migrations();
+        let connect = |migrations: bool, custom_migrations| {
+            let db_opt = db_opt.clone();
+            async move {
+                {
+                    let cfg: Config<Db> = db_opt.into();
```

Comment:
Previously we pulled `db_opt` out of `Config` and now we are converting back into `Config`. We could just use `db.config()` directly right?

Reply:
there was an ownership error w this so I did this

Reply:
I feel like you could just clone the `Config` and move it into this future but idk, this is fine I guess

Reply:
I think I couldn't derive Clone because of generic parameter